examples/vision/learnable_resizer.py

```diff
@@ -0,0 +1,273 @@
+
+"""
+## Model building utility
+"""
+
+
+def get_model():
+    backbone = tf.keras.applications.DenseNet121(
+        weights=None, include_top=True, classes=2
+    )
+    backbone.trainable = True
+
+    inputs = layers.Input((INP_DIM[0], INP_DIM[1], 3))
+    x = layers.experimental.preprocessing.Rescaling(scale=1.0 / 255)(inputs)
+    x = learnable_resizer(x)
+    outputs = backbone(x)
+
+    return tf.keras.Model(inputs, outputs)
+
+
+"""
+## Compile and train our model with learnable resizer
+"""
+
+model = get_model()
+model.compile(
+    loss=keras.losses.CategoricalCrossentropy(label_smoothing=0.1),
+    optimizer="sgd",
+    metrics=["accuracy"],
+)
+model.fit(train_ds, validation_data=validation_ds, epochs=EPOCHS)
+
+"""
+## Visualize the outputs of the trained visualizer
+"""
+
+learned_resizer = tf.keras.Model(model.input, model.layers[-2].output)
+
+plt.figure(figsize=(10, 10))
+for i, image in enumerate(sample_images[:9]):
+    ax = plt.subplot(3, 3, i + 1)
+    image = tf.image.convert_image_dtype(image, tf.float32)
+    resized_image = learned_resizer(image[None, ...])
+    plt.imshow(resized_image.numpy().squeeze())
+    plt.axis("off")
+
+"""
+As we can see the visuals of the images have improved with training. Additionally, you
+can find this repository that shows the benefits of using the resizing module. Below is a
+comparison:
+
+|           Model           	| Number of  parameters (Million) 	| Top-1 accuracy 	|
+|:-------------------------:	|:-------------------------------:	|:--------------:	|
+|   With learnable resizer  	|             7.051717            	|      52.02     	|
+| Without learnable resizer 	|             7.039554            	|      50.3      	|
+
+Note the above-reported models were trained for 10 epochs on 90% of the training set of
+Cats-vs-Dogs unlike this example. Also, note that the increase in the number of
+parameters due to the resizing module is very negligible. You can reproduce these results
+from [this repository](https://github.com/sayakpaul/Learnable-Image-Resizing). To ensure
+that the improvement in the performance is not due to stochasticity, the models were
+trained using the same initial random weights. 
+"""
+
+"""
+## Notes
+
+* To impose shape bias inside the vision models, Geirhos et. al trained them with a
+combination of natural and stylized images. It might be interesting to investigate if
+this learnable resizing module could achieve something similar as the outputs seem to
+discard the texture information. 
+* Through a set of experiments, the authors also verify if the performance improvement is
+***not*** solely due to the increase in the number of model parameters. You are
+encouraged to check those out in the
+[original paper](https://arxiv.org/abs/2103.09950v1). 
+* There is another closely related topic on adaptive image resizing that attempts to
+resize images/feature maps adaptively during training. [EfficientV2](https://arxiv.org/pdf/2104.00298)
+uses this idea. 
```

Comment:
```suggestion
- Note: "et al." <- "et. al"
- Included the link to the first paper, so the user doesn't have to search for it again
- Included the authors of the second and third papers
- "You may find it interesting" <- "It might be interesting" (more sort of active voice this way, talking to the user)
- 80 char per line

```
* To impose shape bias inside the vision models, 
[Geirhos et al.](https://arxiv.org/abs/1811.12231) trained them with a 
combination of natural and stylized images. You may find it interesting to 
investigate if this learnable resizing module could achieve something similar, 
as the outputs appear to discard the texture information. 
* Through a set of experiments, the authors also verify if the performance
improvement is _not_ solely due to the increase in the number of model
parameters. (Check out the [original
paper](https://arxiv.org/abs/2103.09950v1) by Telebi et al.).
* There is another closely related topic on adaptive image resizing that
attempts to resize images/feature maps adaptively during training—
[EfficientV2](https://arxiv.org/pdf/2104.00298) by Tan & Le uses this idea. 
```